pkg/bootstrap/docker/openshift/servicecatalog.go

```diff
@@ -0,0 +1,193 @@
+package openshift
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	aggregatorapi "k8s.io/kube-aggregator/pkg/apis/apiregistration"
+	aggregatorclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/internalclientset/typed/apiregistration/internalversion"
+
+	authzapi "github.com/openshift/origin/pkg/authorization/api"
+
+	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	catalogNamespace        = "service-catalog"
+	catalogService          = "service-catalog"
+	catalogTemplate         = "service-catalog"
+	ServiceCatalogServiceIP = "172.30.1.2"
+)
+
+// InstallServiceCatalog checks whether the service catalog is installed and installs it if not already installed
+func (h *Helper) InstallServiceCatalog(f *clientcmd.Factory, publicMaster, catalogHost string) error {
+	osClient, kubeClient, err := f.Clients()
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	// Grant all users with the edit role, the ability to manage service catalog instance/binding resources.
+	scRule, err := authzapi.NewRule("create", "update", "delete", "get", "list", "watch").Groups("servicecatalog.k8s.io").Resources("instances", "bindings").Rule()
+	if err != nil {
+		return errors.NewError("could not create service catalog resource rule").WithCause(err)
+	}
+
+	editRole, err := osClient.ClusterRoles().Get("edit", metav1.GetOptions{})
+	if err != nil {
+		return errors.NewError("could not get cluster edit role for patching").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	editRole.Rules = append(editRole.Rules, scRule)
+	_, err = osClient.ClusterRoles().Update(editRole)
```

Comment:
Yes, that is correct, based on the code in up.go you will only call InstallServiceCatalog if c.ShouldInitializeData() is true.